ams/optimizer/src/main/java/com/netease/arctic/optimizer/FlinkOptimizerContainer.java

```diff
@@ -72,15 +82,105 @@ protected Map<String, String> doScaleOut(String startUpArgs) {
 
   @Override
   protected String buildOptimizerStartupArgsString(Resource resource) {
-    String taskManagerMemory = PropertyUtil.checkAndGetProperty(resource.getProperties(),
-        TASK_MANAGER_MEMORY_PROPERTY);
-    String jobManagerMemory = PropertyUtil.checkAndGetProperty(resource.getProperties(), JOB_MANAGER_MEMORY_PROPERTY);
+    Map<String, String> properties = resource.getProperties();
+    Map<String, String> flinkConfig = loadFlinkConfig();
+    Preconditions.checkState(properties != null && !flinkConfig.isEmpty(),
+        "resource properties is null or load flink-conf.yaml failed");
+    Long jobManagerMemory = getMemorySizeValue(properties, flinkConfig, JOB_MANAGER_MEMORY_PROPERTY,
+        JOB_MANAGER_TOTAL_PROCESS_MEMORY);
+    Long taskManagerMemory = getMemorySizeValue(properties, flinkConfig, TASK_MANAGER_MEMORY_PROPERTY,
+        TASK_MANAGER_TOTAL_PROCESS_MEMORY);
     String jobPath = getAMSHome() + "/plugin/optimize/OptimizeJob.jar";
-    long memory = Long.parseLong(jobManagerMemory) + Long.parseLong(taskManagerMemory) * resource.getThreadCount();
-    return String.format("%s/bin/flink run -m yarn-cluster -ytm %s -yjm %s -c %s %s -m %s %s",
-        getFlinkHome(), taskManagerMemory, jobManagerMemory,
-        FlinkOptimizer.class.getName(), jobPath, memory,
-        super.buildOptimizerStartupArgsString(resource));
+    long memory = jobManagerMemory + taskManagerMemory * resource.getThreadCount();
+    return String.format("%s/bin/flink run -m yarn-cluster -ytm %s -yjm %s %s -c %s %s -m %s %s",
```

Comment:
This may not comply with the setting of flink's submit cmd.

Reply:
> This may not comply with the setting of flink's submit cmd.

Why does it not meet the requirements， can you provide a detailed explanation? Thank you。

Reply:
Sorry for this, it's my missread.